src/GraphQL/Types/Scalars/IdGraphType.cs

```diff
@@ -23,31 +23,39 @@ public IdGraphType()
             //    "as `\"4\"`) or integer (such as `4`) input value will be accepted as an `ID`.";
         }
 
-        /// <inheritdoc/>
-        public override object Serialize(object value) => value.ToString();
-
         /// <inheritdoc/>
         public override object ParseLiteral(IValue value) => value switch
         {
             StringValue str => ParseValue(str.Value),
             IntValue num => num.Value,
             LongValue longVal => longVal.Value,
+            BigIntValue bigIntValue => bigIntValue.Value,
             _ => null,
         };
 
-        /// <inheritdoc/>
-        public override object ParseValue(object value) => value.ToString().Trim(' ', '"');
-
         /// <inheritdoc/>
         public override bool CanParseLiteral(IValue value) => value switch
         {
             StringValue _ => true,
             IntValue _ => true,
             LongValue _ => true,
+            BigIntValue _ => true,
+            NullValue _ => true,
             _ => false
         };
 
         /// <inheritdoc/>
-        public override IValue ToAST(object value) => new StringValue(value.ToString());
+        public override object ParseValue(object value) => value switch
+        {
+            int _ => value,
+            long _ => value,
+            _ => value?.ToString()
```

Comment:
I suggest to add many cases for integer numbers and call `ThrowValueConversionError` for default case. I would not convert any object (`DateTime`, `Guid`, `bool`, etc.) into string representation how now it does.

Reply:
Of cource `CanParseValue` should be changed too.

Reply:
Good